### PR TITLE
chore: Update Dependencies

### PR DESCRIPTION
**Pull request by jimmymorales**
## 🚀 Description
The following dependencies were updated:
- Android Gradle Plugin `4.1.3` -> `4.2.1`
- Gradle `6.8.3` -> `7.0.2`
- Kotlin `1.4.31` -> `1.5.0` ❤️ 
- Dokka `1.4.20` -> `1.4.32`
- Androidx Appcompat `1.2.0` -> `1.3.0`
- Androidx Core `1.3.2` -> `1.5.0`
- Androidx Fragment `1.3.2.` -> `1.3.4`
- Coroutines `1.4.3` -> `1.5.0`
- Kotest `4.4.0` -> `4.6.0`
- Ktlint `0.40.0` -> `0.41.0`
- Turbine `0.4.1` -> `0.5.1`
- Detekt `1.16.0` -> `1.17.1`

## ✅ Checklist

- [x] My code follows the code style of this project.

### CODE REVIEW

File: .idea/runConfigurations.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="RunConfigurationProducerService">
+    <option name="ignoredProducers">
+      <set>
+        <option value="com.android.tools.idea.compose.preview.runconfiguration.ComposePreviewRunConfigurationProducer" />
+      </set>
+    </option>
+  </component>
+</project>
```
Comment (jvillad1) [inline .idea/runConfigurations.xml:10]: 😞

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-all.zip
```
Comment (jvillad1) [inline gradle/wrapper/gradle-wrapper.properties:3]: bin?

**Review by jvillad1 — APPROVED** [review]
🤖

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (jimmymorales) [reply gradle/wrapper/gradle-wrapper.properties:3]: I think we should stick to using `all`, the difference is that `all` includes source, documentation and examples. The size difference is not that big. 🤷‍♂️

Comment (jvillad1) [reply gradle/wrapper/gradle-wrapper.properties:3]: 🇫🇮

**Pull request merged**